### ISSUE TITLE
CDIoU-CDIoUloss

### ISSUE THREAD

**Issue by hujunchao** [ultralytics/yolov5#2595]
## ❔Question
Can anyone use this mothed in YOLOV5?
author says
 "CDIoU and CDIoU loss have
different excellent performances in several models such as
Faster R-CNN, YOLOv4, RetinaNet and ATSS. There is a
maximum AP improvement of 1.9% and an average AP of
0.8% improvement on MS COCO dataset, compared to traditional evaluation-feedback modules.
"
## Additional context
https://arxiv.org/pdf/2103.11696.pdf
https://github.com/Alan-D-Chen/CDIoU-CDIoUloss/tree/5b911bc2cfe39c671a4e514d06ea73ad1f53292e

**Comment by glenn-jocher** [comment]
@hujunchao seems interesting:
<img width="343" alt="Screenshot 2021-03-25 at 14 35 53" src="https://user-images.githubusercontent.com/26833433/112481560-6deff880-8d77-11eb-9e0b-f71ac091089e.png">


Although YOLO performance improvement was minor (41.2 -> 41.4), and the loss function comes with a new Lambda hyperparameter which would need tuning (appears best at L=0.001):
<img width="353" alt="Screenshot 2021-03-25 at 14 37 00" src="https://user-images.githubusercontent.com/26833433/112481712-95df5c00-8d77-11eb-96c6-e1de0a48491c.png">

If you'd like to try implementing this yourself, the right location to introduce this CDIoU computation would be in bbox_iou() alongside the existing IoU, C/D/GIoU calculations, and then adding a boolean to use it during training. If you make this implementation please consider submitting a PR so everyone else can test it as well!
https://github.com/ultralytics/yolov5/blob/ad05e37d99bd1b86f7223540ad93381b8269d75c/utils/general.py#L337-L380

**Comment by cszer** [comment]
Also saw this paper, will try it few weeks later

**Comment by John-chen521** [comment]
so，how about SCA Loss
